code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm

```diff
@@ -26,15 +27,19 @@
 	var/OldLoc = loc
 	if(NewLoc && !istype(NewLoc, /turf/open/indestructible/reebe_void))
 		var/turf/T = get_turf(NewLoc)
-		for(var/obj/effect/blessing/B in T)
+		if (locate(/obj/effect/blessing, T))
 			if(last_failed_turf != T)
 				T.visible_message("<span class='warning'>[T] suddenly emits a ringing sound!</span>", ignore_mob = src)
 				playsound(T, 'sound/machines/clockcult/ark_damage.ogg', 75, FALSE)
 				last_failed_turf = T
-			to_chat(src, "<span class='warning'>This turf is consecrated and can't be crossed!</span>")
+			if ((REALTIMEOFDAY - lastWarning)) >= 30) 
+				lastWarning = REALTIMEOFDAY
+				to_chat(src, "<span class='warning'>This turf is consecrated and can't be crossed!</span>")
 			return
 		if(!GLOB.ratvar_awakens && istype(get_area(T), /area/chapel))
-			to_chat(src, "<span class='warning'>The Chapel is hallowed ground under a heretical deity, and can't be accessed!</span>")
+			if ((REALTIMEOFDAY - lastWarning)) >= 30) 
+				lastWarning = REALTIMEOFDAY
+				to_chat(src, "<span class='warning'>The Chapel is hallowed ground under a heretical deity, and can't be accessed!</span>")
```

Comment:
@Cyberboss this is such a common pattern, I feel like there should be a component for it.

@DaxDupont that's not a criticism of this change, it's fine